### PR TITLE
feat: nuxt unit tests

### PR DESCRIPTION
**Pull request by maxiroellplenty**
## Why:

Closes: #ID

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/__tests__/fixtures/product.mock.ts [hunk 1]
```diff
@@ -0,0 +1,348 @@
+export const ProductMock = {
```
Comment (ksted) [inline apps/web/__tests__/fixtures/product.mock.ts]: **thought**: Nothing we have to change right now, but I think in the end we'll have to be a bit more specific with the names of the mocks, especially when it comes to products. I imagine we'll want to have something like `SimpleProductMock` that only contains the bare minimum of data required, but then also others like `ProductWithOrderPropertiesMock` for different use cases.

File: apps/web/__tests__/fixtures/product.mock.ts [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+        "hasManufacturer": false,
+        "isSalable": true,
+        "isSalableAndActive": true,
+        "hasActiveChildren": false
+    },
+    "images": {
+        "all": [
+            {
+                "names": [],
+                "url": "https://cdn02.plentymarkets.com/mevofvd5omld/item/images/157/full/gear-headphones-01.png",
```
Comment (ksted) [inline apps/web/__tests__/fixtures/product.mock.ts]: **question**: Do we have to care about the URLs still pointing to real assets? Or would it be prudent of us to have a generic placeholder for these cases?

File: apps/web/components/AttributeSelect/AttributeSelect.vue [hunk 3]
```diff
@@ -27,43 +27,53 @@
...
 const router = useRouter();
 const route = useRoute();
 
-const transformProductData = (product: any) => {
-  const { attributes, variations } = product.variationAttributeMap;
-
-  return {
-    attributes: attributes.map(({ attributeId, name, values }: Attribute) => ({
-      attributeId,
+const transformProductData = (product: Product): TransformedProduct => {
```
Comment (ksted) [inline apps/web/components/AttributeSelect/AttributeSelect.vue:40]: **thought**: This is for a different PR to handle, if at all, but shouldn't these calculations happen in a composable?

File: apps/web/components/CategoryFilters/__tests__/Filter.spec.ts [hunk 4]
```diff
@@ -1,22 +1,39 @@
 import { mount } from '@vue/test-utils';
 import Filter from '~/components/CategoryFilters/Filter.vue';
 
-/*
 describe('<Filter />', () => {
   it('should render component', () => {
     const { getByTestId } = mount(Filter, {
       props: {
         facet: {
```
Comment (ksted) [inline apps/web/components/CategoryFilters/__tests__/Filter.spec.ts:8]: **question**: Why did you decide against using the `FacetMock` here?

File: apps/web/components/CheckoutAddress/__tests__/CheckoutAddress.spec.ts [hunk 5]
```diff
@@ -3,6 +3,7 @@ import CheckoutAddress from '~/components/CheckoutAddress/CheckoutAddress.vue';
 import { AddressType } from '@plentymarkets/shop-api';
 
 describe('<CheckoutAddress />', () => {
+
```
Comment (ksted) [inline apps/web/components/CheckoutAddress/__tests__/CheckoutAddress.spec.ts]:
```suggestion
```
**todo**: Remove inconsistent line break.

File: apps/web/vitest.config.ts [hunk 6]
```diff
@@ -1,16 +1,17 @@
-import { defineVitestConfig } from 'nuxt-vitest/config';
+import { defineVitestConfig } from '@nuxt/test-utils/config';
 
 export default defineVitestConfig({
+  // any custom Vitest config you require
```
Comment (ksted) [inline apps/web/vitest.config.ts]:
```suggestion
```
**todo**: Remove useless comment.

File: apps/web/vitest.config.ts [hunk 7]
```diff
@@ -1,16 +1,17 @@
...
     globals: true,
-    setupFiles: './vue-test-utils.extend.js',
+    clearMocks: true,
+    setupFiles: './vitest.config.setup.ts',
     include: ['**/*/?(*.)+(spec|test).[jt]s'],
-    onConsoleLog: (log) => {
+    onConsoleLog: (log): any => {
       // Silence logs coming from vue <Suspense> is experimental, and stdout | unknown component before it
       if (log.includes('<Suspense')) return false;
       return log;
```
Comment (ksted) [inline apps/web/vitest.config.ts]:
```suggestion
    onConsoleLog: (log): false | void => {
      return silenceLogsFromSuspenseComponent(log);
```
**suggestion**: Specify return type and extract check to replace the inline comment.

```ts
const silenceLogsFromSuspenseComponent = (log: string): false | void => {
  if (log.includes('<Suspense')) return false;
};
```

File: apps/web/vitest.config.setup.ts [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+};
+config.global.stubs = {
+  RouterLink: true,
+  NuxtImg: true,
+  'i18n-t': true,
+};
+
+import { mockComponent, mockNuxtImport } from '@nuxt/test-utils/runtime';
+import { CategoryTreeMock } from './__tests__/fixtures/category-tree.mock';
+import { ShippingCountriesMock } from './__tests__/fixtures/shipping-countries.mock';
```
Comment (ksted) [inline apps/web/vitest.config.setup.ts]: **question**: Is it proper that we spread imports throughout the file?

File: apps/web/vitest.config.setup.ts [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+mockNuxtImport('useActiveShippingCountries', () => {
+  return () => {
+    return {
+      data: computed(() => ShippingCountriesMock.data),
+      getActiveShippingCountries: () => {
+        return vi.fn().mockResolvedValue(ShippingCountriesMock.data);
+      },
+    };
+  };
+});
```
Comment (ksted) [inline apps/web/vitest.config.setup.ts:78]: **question**: Where do we need this mock? Can we just use it in the tests that actually interact with shipping countries?

**Review by ksted — COMMENTED** [review]
1. You've removed a few instances of async/await throughout the app. Can you explain a bit why these changes are necessary?
2. We should discuss where to store mocks. You've put yours into `__tests__/fixtures`, but we also have a couple in `mocks`. As a third option, Vitest suggests using `__mocks__` [at least once](https://vitest.dev/config/#deps-moduledirectories) in their documentation.

File: apps/web/components/CategoryFilters/__tests__/Filter.spec.ts (hunk 4 above)
Comment (maxiroellplenty) [reply apps/web/components/CategoryFilters/__tests__/Filter.spec.ts:8]: There was no FacetMock before :D Also its a different facet/Filter

File: apps/web/vitest.config.setup.ts (hunk 8 above)
Comment (maxiroellplenty) [reply apps/web/vitest.config.setup.ts]: no I will move them

File: apps/web/vitest.config.setup.ts (hunk 9 above)
Comment (maxiroellplenty) [reply apps/web/vitest.config.setup.ts:78]: That's the problem, we can't use the mockNuxtImport in the test where we need it. This needs more research.

File: apps/web/vitest.config.ts (hunk 7 above)
Comment (maxiroellplenty) [reply apps/web/vitest.config.ts]: will change it

File: apps/web/components/AttributeSelect/AttributeSelect.vue (hunk 3 above)
Comment (maxiroellplenty) [reply apps/web/components/AttributeSelect/AttributeSelect.vue:40]: Created follow up task: https://dev.azure.com/plentymarkets/plentymarkets/_workitems/edit/91558

File: apps/web/__tests__/fixtures/product.mock.ts (hunk 2 above)
Comment (maxiroellplenty) [reply apps/web/__tests__/fixtures/product.mock.ts]: I think its not important for now

File: apps/web/__tests__/fixtures/product.mock.ts (hunk 1 above)
Comment (maxiroellplenty) [reply apps/web/__tests__/fixtures/product.mock.ts]: Yes thats the plan 👍

**Comment by maxiroellplenty** [comment]
> 1. You've removed a few instances of async/await throughout the app. Can you explain a bit why these changes are necessary?
> 2. We should discuss where to store mocks. You've put yours into `__tests__/fixtures`, but we also have a couple in `mocks`. As a third option, Vitest suggests using `__mocks__` [at least once](https://vitest.dev/config/#deps-moduledirectories) in their documentation.

1. Our test cant handle async setup methods as it turns out. Thats an issue! Good thing is for now all the awaits that I removed were unnecessary. Address form works without await. The second one was just by mistake there was no reason for an await/async there.
2. Yes I will rename the folder/ create a new __mocks__ folder and move the files

File: apps/web/vitest.config.setup.ts (hunk 9 above)
Comment (ksted) [reply apps/web/vitest.config.setup.ts:78]: Definitely. Maybe we can figure out a difference between our cases and the [nuxt/utils examples](https://github.com/search?q=repo%3Anuxt%2Ftest-utils+mocknuxtimport+path%3A%2F%5Eexamples%5C%2F%2F&type=code), if you haven't already looked at those.

**Review by ksted — APPROVED** [review]

**Pull request merged**